influxdb/2.4/alpine/entrypoint.sh

```diff
@@ -150,34 +158,41 @@ function influxd::config::get()
   # it is possible that variable was intentionally emptied; therefore, this
   # returns nothing when empty.
   value="$(table::get "${primary_key}" ${COLUMN_ENVIRONMENT})"
-  if [[ "${!value+x}" ]] ; then
-    echo "${!value:-}" && return
+  if [[ ${!value+x} ]] ; then
+    echo "${!value}" && return
   fi
 
-  # Finally, search the configuration files. `yq` is required as the format
-  # could be json, toml, yaml, etc. If the corresponding value could not be
-  # located within the configuration files, provide the default value. This
-  # default is what `influx server-config` would display if recently
-  # initialized with `influx setup`.
-  default="$(table::get "${primary_key}" "${COLUMN_DEFAULT}")"
-
-  if [[ "${INFLUXD_CONFIG_PATH}" == *toml ]]
+  #
+  # Parse Value from Configuration
+  #
+
+  case ${INFLUXD_CONFIG_PATH,,} in
+    *.toml)       local influxd_config_format=toml ;;
+    *.json)       local influxd_config_format=json ;;
+    *.yaml|*.yml) local influxd_config_format=yaml ;;
+  esac
+
+  # Unfortunately, dasel does not provide a method to test for the existence
+  # of a particular key. However, it returns with an error if it attempts to
+  # delete a nonexistent key. Since dasel will automatically update the
+  # configuration file, it is passed via `STDIN` to avoid this behavior.
+  if dasel -r "${influxd_config_format}" delete "${primary_key}" <"${INFLUXD_CONFIG_PATH}" &>/dev/null
   then
-    # There are two yq projects. Unfortunately, only the python version can
-    # process toml documents and this is the go implementation. Downloading
-    # python libraries would increase the docker image size considerably.
-    # Fortunately, processing toml as props extracts most information.
-    value="$(yq -p props eval ".\"${primary_key}\" // \"${default}\"" "${INFLUXD_CONFIG_PATH}")"
+    # Unfortunately, dasel has inconsistent string displaying behavior. For
+    # yaml and toml, it will display the value without double-quotes unless
+    # the string is empty. However, for json, it always with double-quotes.
+    # This always strips the first leading and trailing double-quotes, as
+    # they are almost always not required.
```

Comment:
Is there any chance the "almost always" comes back to bite us at some point?

Reply:
I can't say with 100% certainty, but I think that the odds are very low. It would have to parse a string that is intentionally surrounded in double quotes. I don't think any configuration values require this at the moment.